core/templates/pages/exploration-player-page/services/current-interaction.service.ts

```diff
@@ -70,8 +72,12 @@ export class CurrentInteractionService {
     CurrentInteractionService.onSubmitFn = onSubmit;
   }
 
+  // The interaction can pass in null if it
+  // does not use the progress nav's submit button.
```

Comment:
I'm not sure I understand this? How os this connected to nav's submit button?

Reply:
If we take the example of interaction MultipleChoiceInput. If the user submits the solution by clicking on the navbars submit buttons then it cannot pass null as we have a check for that (file :oppia-interactive-multiple-choice-input.component.ts, line :178). But if the solution is submitted without that (i.e the function grabs the solution by itself) then the interaction can pass null value.
PTAL at comment present at line 85 of the same service.